server/pypi/make.sh

```diff
@@ -1,57 +1,285 @@
+# pyzmq
+# rawpy
+# rpi-gpio
+# scikit-image
+# scikit-learn
+# scipy
+# sentencepiece
+# shapely
+# soundfile
+# spacy
+# ta-lib
+# tensorflow
+# tflite-runtime
+# thinc
+# tokenizers
+# torch
+# torchvision
+# xgboost
+
+# parse command line
+
+POSITIONAL=()
+while [[ $# -gt 0 ]] ; do
+  key="${1}"
+
+  case ${key} in
+    --help)
+      echo "Usage: make.sh options
+
+      options
+              --package package
+              --skip-deps
+              --skip-envs
+              --year year"
+      exit 0
+      ;;
+    --package)
+      PACKAGES=${2}
+      shift
+      shift
+      ;;
+    --skip-deps)
+      BUILD_DEPS="0"
+      shift
+      ;;
+    --skip-envs)
+      BUILD_ENVS="0"
+      shift
+      ;;
+    --year)
+      YEAR=${2}
+      shift
+      shift
+      ;;
+    *)
+      POSITIONAL+=("$1") # save it in an array for later
+      shift
+    ;;
+  esac
+done
+
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+# setup clean conda environments and toolchains for each Python versions
+
+eval "$(command conda 'shell.bash' 'hook')"
+conda activate base
+
+if [ "${BUILD_ENVS}" == "1" ]; then
+  rm -rf "${LOGS}" "${TOOLCHAINS}"
+  curl --silent --location "https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-macos-universal.tar.gz" --output cmake.tar.gz
+  mkdir -p "${TOOLCHAINS}"
+  tar -xzf cmake.tar.gz --directory "${TOOLCHAINS}"
+  mv "${TOOLCHAINS}/cmake-${CMAKE_VERSION}"*/CMake.app "${TOOLCHAINS}"
+  rm -rf "${TOOLCHAINS}/cmake-${CMAKE_VERSION}"* cmake.tar.gz
+
+  for PYTHON_VERSION in ${PYTHON_VERSIONS}; do
+    rm -rf "$(conda info | grep 'envs directories' | awk -F ':' '{ print $2 }' | sed -e 's/^[[:space:]]*//')/${CONDA_ENV:?}"
+    conda create -y --name "${CONDA_ENV}-${PYTHON_VERSION}" "python==${PYTHON_EXACT_VERSIONS[${PYTHON_VERSION}]}"
+    conda activate "${CONDA_ENV}-${PYTHON_VERSION}"
+    pip3 install -r requirements.txt
+    curl --silent --location "$(curl --silent -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" --location https://api.github.com/repos/beeware/Python-Apple-support/releases | grep browser_download_url | grep iOS | grep "${PYTHON_VERSION}" | head -1 | awk '{ print $2 }' | tr -d '"')" --output "python-apple-support-${PYTHON_VERSION}.tar.gz"
+    mkdir -p "${TOOLCHAINS}/${PYTHON_VERSION}" "${LOGS}/${PYTHON_VERSION}"
+    tar -xzf "python-apple-support-${PYTHON_VERSION}.tar.gz" --directory "${TOOLCHAINS}/${PYTHON_VERSION}"
+    rm -rf "python-apple-support-${PYTHON_VERSION}.tar.gz" "${LOGS:?}/${PYTHON_VERSION:?}"/*
+  done
+fi
+
+# build dependencies
+
+if [ "${BUILD_DEPS}" == "1" ]; then
+  mkdir -p "${LOGS}/deps"
+  pushd "${TOOLCHAINS}"
+
+  if ! [ -d "${PYTHON_APPLE_SUPPORT}" ]; then
+    git clone https://github.com/beeware/Python-Apple-support.git
+  fi
+
+  conda activate "${CONDA_ENV}-3.10"
+  pushd "${PYTHON_APPLE_SUPPORT}"
+  git checkout 3.10
+  make libFFI-wheels
+  # TODO: make OpenSSL-wheels
```

Comment:
It does build but the wheel doesn't complete.
`
# Copy distributable content
cp -r /Users/yves/Downloads/chaquopy/server/pypi/support/Python-Apple-support/install/iOS/iphonesimulator.x86_64/openssl-3.0.5/include /Users/yves/Downloads/chaquopy/server/pypi/support/Python-Apple-support/install/iOS/iphonesimulator.x86_64/openssl-3.0.5/wheel/opt/include
cp: /Users/yves/Downloads/chaquopy/server/pypi/support/Python-Apple-support/install/iOS/iphonesimulator.x86_64/openssl-3.0.5/include: No such file or directory
make: *** [wheels/dist/openssl/openssl-3.0.5-1-py3-none-ios_12_0_iphonesimulator_x86_64.whl] Error 1
`

Reply:
I wonder if it's a Makefile issue with a missing (or incorrect) dependency. I haven't thoroughly tested each Makefile target; I'm normally running a top level `make all`, and then running `make X-wheels` as an afterthought. I've tried running locally (in an admittedly pre-built environment), and I noticed that `make OpenSSL-wheels` triggered a Python build, which would seem to suggest the build chain isn't exactly "clean" - but the build did complete, including creating the wheels. 

Related - one of the items on my todo list is beeware/Python-Apple-support#169 - reworking the Python support package so that each of the dependencies can be built independently. This will likely involve introducing separate repositories for libFFI, BZip2, XZ and OpenSSL so that we can build those dependencies independently of Python (and have independent publication processes as well).

Reply:
Will try to build all first and test again. Was just trying to save time.